v3/src/models/data/formula-manager.ts

```diff
@@ -270,7 +272,31 @@ export class FormulaManager {
           this.recalculateFormula(formulaId)
         }
       })
-    }, { equals: comparer.structural, fireImmediately: true })
+    }, {
+      equals: comparer.structural,
+      fireImmediately: true,
+      name: "FormulaManager.registerAllFormulas.reaction"
+    })
+  }
+
+  unregisterDeletedFormulas() {
+    this.formulaMetadata.forEach((metadata, formulaId) => {
+      const { dataSetId, attributeId } = this.getFormulaMetadata(formulaId)
+      const dataSet = this.dataSets.get(dataSetId)
+      if (!dataSet) {
+        this.unregisterFormula(formulaId)
+        return
+      }
+      if (!dataSet.attrFromID(attributeId)) {
+        this.unregisterFormula(formulaId)
+        return
+      }
+      // In 99% of cases, the two `if` statements above will be sufficient to detect deleted formulas. However, this
+      // could serve as the ultimate check in case a formula instance was deleted in a different manner.
+      if (!isAlive(metadata.formula)) {
+        this.unregisterFormula(formulaId)
+      }
```

Comment:
>I remember that during one of the recent arch meetings, you mentioned some issues with `isAlive`, but I can't remember exactly what that was.

It was just a recommendation to not use it as a primary control flow mechanism but rather as a means of detecting that something isn't being handled optimally, as you are doing here. Toward that end, I'd suggest adding a `console.warn`, something like:
```suggestion
      if (!isAlive(metadata.formula)) {
        console.warn("FormulaManager.unregisterDeletedFormulas unregistering a defunct formula that was not detected by the usual means.")
        this.unregisterFormula(formulaId)
      }
```